core/commands/repo.go

```diff
@@ -65,24 +66,11 @@ order to reclaim hard disk space.
 
 		go func() {
 			defer close(outChan)
-			var errors []error
-			for res := range gcOutChan {
-				if res.KeyRemoved != nil {
-					outChan <- &corerepo.KeyRemoved{res.KeyRemoved}
-				}
-				if res.Error != nil {
-					errors = append(errors, res.Error)
-				}
-			}
-			switch len(errors) {
-			case 0:
-				return
-			case 1:
-				res.SetError(errors[0], cmds.ErrNormal)
-				return
-			default:
-				res.SetError(corerepo.NewMultiError(errors...), cmds.ErrNormal)
-				return
+			err := corerepo.CollectResult(gcOutChan, func(k *cid.Cid) {
```

Comment:
Though you probably want to select with a context here to make sure there are no accidental hangs